debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerQueryResultProcessor.java

```diff
@@ -268,6 +268,42 @@ private boolean hasNext(ResultSet resultSet) throws SQLException {
         return false;
     }
 
+    private Table dispatchSchemaChangeEventAndGetTableForNewMonitoredTable(TableId tableId) throws SQLException {
+        try {
+            LOGGER.info("Table {} is new and will be monitored.", tableId);
+            offsetContext.event(tableId, Instant.now());
+            dispatcher.dispatchSchemaChangeEvent(
+                    tableId,
+                    new OracleSchemaChangeEventEmitter(
+                            connectorConfig,
+                            offsetContext,
+                            tableId,
+                            tableId.catalog(),
+                            tableId.schema(),
+                            getTableMetadataDdl(tableId),
```

Comment:
Why do we need to do this out-of-bands call here?

Reply:
Using the same connection lead to the following exception:
```
java.sql.SQLException: Closed Resultset: next
	at oracle.jdbc.driver.InsensitiveScrollableResultSet.ensureOpen(InsensitiveScrollableResultSet.java:133)
	at oracle.jdbc.driver.InsensitiveScrollableResultSet.next(InsensitiveScrollableResultSet.java:428)
	at io.debezium.connector.oracle.logminer.LogMinerQueryResultProcessor.hasNext(LogMinerQueryResultProcessor.java:266)
	at io.debezium.connector.oracle.logminer.LogMinerQueryResultProcessor.processResult(LogMinerQueryResultProcessor.java:100)
	at io.debezium.connector.oracle.logminer.LogMinerStreamingChangeEventSource.execute(LogMinerStreamingChangeEventSource.java:187)
	at io.debezium.pipeline.ChangeEventSourceCoordinator.streamEvents(ChangeEventSourceCoordinator.java:159)
	at io.debezium.pipeline.ChangeEventSourceCoordinator.lambda$start$0(ChangeEventSourceCoordinator.java:122)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```
I'll take a look over tomorrow or Monday to see why that is.  I'm not sure whether by us running an `ALTER SESSION` that if that would invalidate open result-sets or if there is something else that leads to the ResultSet being closed.

Reply:
Ok, but why the need for any call _at all_? Shouldn't the table already be part of the metadata at this point, e.g. by parsing a CREATE TABLE DDL earlier on?

Reply:
What about the case where we deploy a connector and the `table.include.list` includes TABLEA and TABLEB but only TABLEA is included in the `snapshot.include.collection.list`.  In this scenario, the schema is only pre-filled with TABLEA and not TABLEB so when we detect any DML for TABLEB that is included by the filters, we need to add this table to the schema before we can emit events for it; which is what this is to solve.

Reply:
Ok, I see where you're coming from. In think in this case, the connector should still capture and historise the schema of all tables matching the filter config, but only emit data change events for those matching `snapshot.include.collection.list`, so that it'd then know the schema when receiving a change for a table not on that list during streaming. Can you check whether that's the case for MySQL for instance?

Reply:
I'll check MySQL but I know for SQL Server that most definitely wasn't the case.

Reply:
Any news on this one? I still believe we should strive to avoid this out-of-bands call.

Reply:
So I compared the legacy to the common-framework implementation of MySQL and you're correct that the legacy impl does in fact capture the table schema for all included tables but only generates snapshot events for the tables in the special list setting; however the common framework implementation behaves like Oracle and SQL Server.

With that in mind, I think we need to revisit how this works across all connectors and make sure this works uniformly like the old MySQL legacy implementation as they seems to be the standard this was to be based upon.  Would you agree?

Reply:
Yes, I would agree. Also pinging @jpechane for awareness.

Reply:
That said, does the SQL Server connector also do an out-of-bands call? If so, we may decide to go with the proposed implementation here for now and do this rework as a follow-up for CR1.

Reply:
It does, see [SqlServerStreamingChangeEventSource.java#L381](https://github.com/debezium/debezium/blob/master/debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java#L381).

Reply:
Ok, I see.

So @jpechane, @Naros: WDYT, should we merge this PR then as is, or rather tackle the proper fix, including SQL Server then? The feature adds small-ish value and isn't urgent, so I probably would rather do it right from the beginning. OTOH, it could be used already by folks safely who can rule out any concurrent schema changes and it's always nice to get something out of our way. I'm on the fence :)

Reply:
So to be precise: I vote for merging this now (with proper fix scheduled for CR1), simple majority of our three votes decides :)

Reply:
@gunnarmorling  I a,m fine with merging this as is and then moving/adding support to common connector code of necessary for the other connectors. Just please bear in mind that in case of SQL Server the out-of-bands call is unavodable as there is only way how we can recive the tabe structure.

Reply:
Ok, let's do it that way then.